lib/keen/client/publishing_methods.rb

```diff
@@ -122,6 +122,7 @@ def beacon_url(event_collection, properties)
       def redirect_url(event_collection, properties, redirect_url)
         require 'open-uri'
         encoded_url = URI::encode(redirect_url)
+          .gsub("&", "%26") # Keen API requires & to be escaped, which URI::encode does not do
```

Comment:
So, ruby 1.8.7 doesn't like this syntax. Can you move the .gsub onto the same line, that should make the tests pass!